Cabal/src/Distribution/Simple/Build.hs

```diff
@@ -114,6 +114,31 @@ build pkg_descr lbi flags suffixes = do
 
   internalPackageDB <- createInternalPackageDB verbosity lbi distPref
 
+  -- Before the actual building, dump out build-information.
+  -- This way, if the actual compilation failed, the options have still been
+  -- dumped.
+  when shouldDumpBuildInfo $ do
```

Comment:
@DanielG, I moved this before building, afaict this allows us to extract the information, even if there is a compilation error. You think this is fine, too?